code/controllers/subsystem/supply_shuttle.dm

```diff
@@ -473,11 +473,14 @@ var/datum/subsystem/supply_shuttle/SSsupply_shuttle
 			for(var/typepath in CF.contains)
 				if(!typepath)
 					continue
+				var/this_amount = CF.amount
 				var/atom/B2 = new typepath(CF.associated_crate)
 				if(istype(B2,/obj/item/stack))
 					var/obj/item/stack/S = B2
-					if(CF.amount && S.amount)
-						S.amount = CF.amount < S.max_amount ? CF.amount : S.max_amount // Just cap it here
+					if(this_amount && S.amount)
+						S.amount = min(this_amount,S.max_amount)
+						if(this_amount > S.max_amount)
+							this_amount -= S.max_amount //leave the count over for the next pile until the last has the remainder under it
```

Comment:
Does this function as intended? When the loop restarts, this_amount is reset to CF.amount's value and is lost. This could be fixed by moving the var/this_amount definition outside of the for loop. That way, using the bug report as an example, the 108 diamonds passed to this proc as 3 stacks of diamonds with 108 amount will properly have the 58 then 8 this_amounts passed along.

Reply:
thanks for spotting this, all the stuff i tested got less than this and worked fine. fixed